appinterface/projection/crossfire/crossfire.go

```diff
@@ -691,19 +692,33 @@ func (projection *Crossfire) computeTxSentRank(
 	// Using dummy list for development
 	participantsList, errGetRequest := projection.Client.Participants()
 	if errGetRequest != nil {
-		return fmt.Errorf("[error] Getting Participants list %w", errGetRequest)
+		return fmt.Errorf("[Crossfire] error getting Participants list %w", errGetRequest)
+	}
+
+	participantsMap := make(map[string]Participant)
+	for _, participant := range *participantsList {
+		participantsMap[participant.PrimaryAddress] = participant
 	}
 
 	// Get All Tx Count Sorted
-	dbParticipantWithCountList, errDbCount := crossfireValidatorStatsView.FindAllLike(constants.TOTAL_TX_SENT_PREFIX)
+	dbTxSendersWithCountList, errDbCount := crossfireValidatorStatsView.FindAllLike(constants.TOTAL_TX_SENT_PREFIX)
 	if errDbCount != nil {
-		return fmt.Errorf("[error] Database Participant With Total Count %w", errDbCount)
+		return fmt.Errorf("[Crossfire] error Database Participant With Total Count %w", errDbCount)
 	}
-	if len(dbParticipantWithCountList) < 1 {
+	if len(dbTxSendersWithCountList) < 1 {
 		// No Participants to Rank
 		return nil
 	}
 
+	// Filter only participants from tx senders list
+	dbParticipantWithCountList := make([]view.CrossfireValidatorsStatsRow, 0)
+	for _, dbTxSender := range dbTxSendersWithCountList {
+		dbParticipantPrimaryAddress := strings.Split(dbTxSender.Key, constants.DB_KEY_SEPARATOR)[1]
+		if _, ok := participantsMap[dbParticipantPrimaryAddress]; ok {
+			dbParticipantWithCountList = append(dbParticipantWithCountList, dbTxSender)
+		}
+	}
+
```

Comment:
@cdc-Hitesh Found a bug regarding the ranking logic, please help to verify if this has fixed it.

Reply:
Looks good.